pkg/api/defaults-custom-cloud-profile.go

```diff
@@ -22,61 +22,78 @@ type CustomCloudProfileDefaultsParams struct {
 	IsScale   bool
 }
 
+func getAzureStackFQDNSuffix(portalUrl, location string) string {
+	azsFQDNSuffix := strings.Replace(portalUrl, fmt.Sprintf("https://portal.%s.", location), "", -1)
+	azsFQDNSuffix = strings.TrimSuffix(azsFQDNSuffix, "/")
+
+	return azsFQDNSuffix
+}
+
 func (cs *ContainerService) setCustomCloudProfileDefaults(params CustomCloudProfileDefaultsParams) error {
 	p := cs.Properties
-	if p.IsAzureStackCloud() {
+	if p.IsCustomCloudProfile() {
 		p.CustomCloudProfile.AuthenticationMethod = helpers.EnsureString(p.CustomCloudProfile.AuthenticationMethod, ClientSecretAuthMethod)
 		p.CustomCloudProfile.IdentitySystem = helpers.EnsureString(p.CustomCloudProfile.IdentitySystem, AzureADIdentitySystem)
-		p.CustomCloudProfile.DependenciesLocation = DependenciesLocation(helpers.EnsureString(string(p.CustomCloudProfile.DependenciesLocation), AzureStackDependenciesLocationPublic))
+		p.CustomCloudProfile.DependenciesLocation = DependenciesLocation(helpers.EnsureString(string(p.CustomCloudProfile.DependenciesLocation), AzureCustomCloudDependenciesLocationPublic))
 		err := cs.SetCustomCloudProfileEnvironment()
 		if err != nil {
 			return fmt.Errorf("Failed to set environment - %s", err)
 		}
-		err = p.SetAzureStackCloudSpec(AzureStackCloudSpecParams(params))
+		err = p.SetCustomCloudSpec(AzureCustomCloudSpecParams(params))
 		if err != nil {
 			return fmt.Errorf("Failed to set cloud spec - %s", err)
 		}
 	}
 	return nil
 }
 
-// SetCustomCloudProfileEnvironment retrieves the endpoints from Azure Stack metadata endpoint and sets the values for azure.Environment
+// SetCustomCloudProfileEnvironment retrieves the endpoints from metadata endpoint (when required) and sets the values for azure.Environment
 func (cs *ContainerService) SetCustomCloudProfileEnvironment() error {
 	p := cs.Properties
-	if p.IsAzureStackCloud() {
+	if p.IsCustomCloudProfile() {
```

Comment:
Because the endpoints are well-known, I would suggest to pre-define the azure.Environment in aks engine code base and directly use it get the  azure.Environment based on the environment name.
Azure Stack has dynamic endpoints. we are forced to using the logic to populate the endpoints.
1. The logic to format the endpoints are different between Azure Stack and LX cloud.
2. Not all endpoints are filled since not all services are available on Azure Stack.

Reply:
We cannot add any LX-specific URLs in a public repo (got a confirmation from LX folks) :(

Reply:
is this case, Please exam the logic to format the endpoints to meet the LX requirement.